src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,104 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int PRIME_NUMBER_17 = 17;
+    private static final int PRIME_NUMBER_31 = 31;
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_LOAD_FACTOR * DEFAULT_INITIAL_CAPACITY);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resizeCountOfBuckets();
+        }
+        int indexOfBucket = getIndexOfBucket(key);
+        if (table[indexOfBucket] == null) {
+            table[indexOfBucket] = new Node<>(key, value, null);
+            size++;
+        } else {
+            checkingOfTheBucket(table[indexOfBucket], key, value);
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int indexOfBucket = getIndexOfBucket(key);
+        Node<K, V> entry = table[indexOfBucket];
+        while (entry != null) {
+            if (Objects.equals(entry.key, key)) {
+                return entry.value;
+            }
+            entry = entry.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getHash(K key) {
```

Comment:
let's arrange your methods in the order they are called. I mean, this method should be declared after your getIndexOfBucket()